src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,43 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.Message;
+import codeu.model.data.RoutingEngine;
+
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+	
+
+public class parsingEngine {
+	/* 
+	 * Here, regex is a better way to parse bot commands rather than matching them with some keywords.
+	 * Regex can parse varying bot request according to pattern. 
+	 * For example, the following regex can parse "/time", "/  time" or "/time    " as "time". 
+	 * 
+	 */
+	
+	static final String commandRegex = "^/(?P<command_name>\\w+)(?:\\s+(?P<argument>\\w+))?\\s*$";
+	
+	// compile the Regex command
+	 static final Pattern pattern = Pattern.compile(commandRegex);
+	
+ 	public void ParseCommands(Message message){ 
```

Comment:
This might actually be able to be `static` since we don't really require any initialization of `RoutingEngine` to call this